packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/TypeFactory.cs

```diff
@@ -10,13 +10,53 @@ namespace Microsoft.Generator.CSharp
 {
     public abstract class TypeFactory
     {
+        private readonly IDictionary<InputModelType, ModelProvider> _models = new Dictionary<InputModelType, ModelProvider>();
+        private readonly IDictionary<InputEnumType, EnumProvider> _enums = new Dictionary<InputEnumType, EnumProvider>();
+        private readonly IDictionary<InputClient, ClientProvider> _clients = new Dictionary<InputClient, ClientProvider>();
```

Comment:
Should ClientProvider live on ScmTypeFactory instead?

Reply:
So this is the part I am not quite sure, do we decide how the customers' plugin works?
If the customer's plugin should always be built on top of the Scm plugin - I think I could leave all the CreateModel/Enum/Client method abstract, and move the caches into ScmTypeFactory. Also this raise another question: do we need to put the default model implementation here as well? Or it should be in Scm project as well?
If the customer's plugin could only be built on top of this basic framework, I think it is very inconvenient if we do not have this cache because the customer will always need this

Reply:
So checking the changes in Michael's PR: https://github.com/microsoft/typespec/pull/3611/files
I think maybe to align, we should move these cache to `ScmTypeFactory` and leave those methods abstract.

Reply:
So I have moved these to `ScmTypeFactory`.